blocks/kanban-board/src/components/board/column/column.tsx

```diff
@@ -0,0 +1,109 @@
+import { SortableContext, useSortable } from "@dnd-kit/sortable";
+import { CSS } from "@dnd-kit/utilities";
+import clsx from "clsx";
+
+import { IconButton } from "../../icon-button/icon-button";
+import { DiscardIcon } from "../../icons/discard-icon";
+import { PlusIcon } from "../../icons/plus-icon";
+import { Card } from "../card/card";
+import {
+  ColumnData,
+  CreateCardCallback,
+  DeleteCardCallback,
+  DeleteColumnCallback,
+  UpdateCardContentCallback,
+  UpdateColumnTitleCallback,
+} from "../types";
+import { EditableColumnTitle } from "./editable-column-title/editable-column-title";
+import styles from "./styles.module.scss";
+
+interface ColumnProps {
+  data: ColumnData;
+  deleteColumn: DeleteColumnCallback;
+  createCard: CreateCardCallback;
+  deleteCard: DeleteCardCallback;
+  updateColumnTitle: UpdateColumnTitleCallback;
+  updateCardContent: UpdateCardContentCallback;
+  readonly?: boolean;
+}
+
+export const Column = ({
+  data,
+  deleteColumn,
+  createCard,
+  deleteCard,
+  updateCardContent,
+  updateColumnTitle,
+  readonly,
+}: ColumnProps) => {
+  const {
+    attributes,
+    listeners,
+    setNodeRef,
+    transform,
+    transition,
+    isDragging,
+    isSorting,
+  } = useSortable({ id: data.id });
+
+  const style = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+    opacity: isDragging ? 0.5 : 1,
+  };
+
+  return (
+    <div
+      className={styles.wrapper}
+      ref={setNodeRef}
+      style={style}
+      {...attributes}
+    >
+      <div
+        className={clsx(
+          styles.titleRow,
+          (isDragging || isSorting) && styles.showHandCursor,
+        )}
+        {...listeners}
+      >
+        <EditableColumnTitle
+          title={data.title}
+          onChange={(val) => updateColumnTitle(data.id, val)}
+          readonly={readonly}
+        />
+        {!readonly && (
+          <IconButton onClick={() => deleteColumn(data.id)}>
+            <DiscardIcon />
+          </IconButton>
+        )}
+      </div>
+      <div className={styles.body}>
+        <SortableContext
+          items={data.cards.map((card) => card.id)}
```

Comment:
https://github.com/hashintel/hash/pull/2234/commits/a3b937870ed6a97f852acabbcf2746edf876ccd0 moved here!

Reply:
This change broke the drag preview.
![bug](https://user-images.githubusercontent.com/37453800/227230650-8e49d09c-9069-4cd9-9b3c-39cb17e88da9.gif)

This is because the recalculation isn't happening every time the order of the cards change, you need to set the dependency array to `data.cards` instead

Reply:
Thanks for noticing this, fixed here https://github.com/hashintel/hash/pull/2234/commits/4daed043eb7ef89b4403fc8f528ce4da83e3ce60